src/components/Checkbox/index.test.tsx

```diff
@@ -8,7 +8,7 @@ describe('Checkbox component', () => {
     const { container } = render(
       <Checkbox label="change callback" disabled name="disabled-test" />
     );
-    const input = screen.getByLabelText('change callback');
+    const input = screen.getByLabelText('change callback') as HTMLInputElement;
```

Comment:
[Conversation] I don't like that we're losing type safety here, but I also don't have a better solution at the moment.

We _could_ extend React Test Library to offer something like `screen.getInputByLabelText`

```typescript
function getInputByLabelText (text): HTMLInputElement {
  const element = getByLabelText(text);
  if (element.nodeName !== 'INPUT') {
    throw new Error('Not an input!');
  }
  
  return element as HTMLInputElement;
}
```

We could even make that more generic to accept the expected node, but I'm wondering if that's more energy than it's worth. I think it depends on how often we need the solution and how easy it is to import and use.

What I'm trying to avoid is people using this `as X` syntax in production code and being bit by it because they didn't realize that the types actually knew something they didn't. Seeing examples like these makes people feel like they have permission to use them elsewhere even if they don't fully understand the implications.

In the example of `getLabelByText`, `input`s are not the only elements that are able to be labeled, and not all of them support `disabled`.

Reply:
Well these tests are a little bit more introspective and that's why they require the casting. Prod code shouldn't need this.

Reply:
I'm going to do a little research here myself to see what people are doing here and then approve this a bit later if people are not using any alternative solutions for this.

Reply:
```suggestion
    const input = screen.getByLabelText('change callback');
```

Reply:
The only reason I did this. It's because otherwise the tests will have type errors. Do you prefer that? If so I can change it back.

Reply:
Sorry didn't catch this comment. 

- Combined with the other suggested adjustments, there should be no further type errors.
- The problem is directly accessing attributes off of elements, rather than using the appropriate expectations from testing library jest-dom custom matchers: https://github.com/testing-library/jest-dom
- I've setup CI to fail if any type errors whatsoever are introduced.